src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -115,67 +115,80 @@ internal sealed class Arguments : TransformInputBase
+                        extractor.AllLengths = _wordFeatureExtractor.AllLengths;
+                        extractor.MaxNumTerms = _wordFeatureExtractor.MaximumNgramsCount;
+                        extractor.Weighting = _wordFeatureExtractor.Weighting;
+                    }
+                    WordFeatureExtractorFactory = extractor;
+                }
+            }
+
+            [TGUI(Label = "Char Gram Extractor")]
+            [Argument(ArgumentType.Multiple, Name = "CharFeatureExtractor", HelpText = "Ngram feature extractor to use for characters (WordBag/WordHashBag).", ShortName = "charExtractor", NullName = "<None>", SortOrder = 12)]
+            internal INgramExtractorFactoryFactory CharFeatureExtractorFactory = new NgramExtractorTransform.NgramExtractorArguments() { NgramLength = 3, AllLengths = false };
+
             /// <summary>
-            /// Whether to use char extractor or not.
+            /// The underlying state of <see cref="CharFeatureExtractorFactory"/> and <see cref="CharFeatureExtractor"/>
             /// </summary>
-            public bool UseCharExtractor { get; set; } = true;
+            private WordBagEstimator.Options _charFeatureExtractor;
+
             /// <summary>
-            /// Whether to use word extractor or not.
+            /// Ngram feature extractor to use for characters (WordBag/WordHashBag).
             /// </summary>
-            public bool UseWordExtractor { get; set; } = true;
-#pragma warning restore MSML_NoInstanceInitializers // No initializers on instance fields or properties
+            public WordBagEstimator.Options CharFeatureExtractor
+            {
+                get { return _charFeatureExtractor; }
+                set
+                {
+                    _charFeatureExtractor = value;
+                    NgramExtractorTransform.NgramExtractorArguments extractor = null;
+                    if (_charFeatureExtractor != null)
+                    {
+                        extractor = new NgramExtractorTransform.NgramExtractorArguments();
+                        extractor.NgramLength = _charFeatureExtractor.NgramLength;
+                        extractor.SkipLength = _charFeatureExtractor.SkipLength;
+                        extractor.AllLengths = _charFeatureExtractor.AllLengths;
+                        extractor.MaxNumTerms = _charFeatureExtractor.MaximumNgramsCount;
+                        extractor.Weighting = _charFeatureExtractor.Weighting;
+                    }
+                    CharFeatureExtractorFactory = extractor;
+                }
+            }
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Normalize vectors (rows) individually by rescaling them to unit norm.", ShortName = "norm", SortOrder = 13)]
+            public NormFunction VectorNormalizer = NormFunction.L2;
+
+            public Options()
+            {
+                WordFeatureExtractor = new WordBagEstimator.Options();
```

Comment:
>WordFeatureExtractor [](start = 16, length = 20)

This will kill the initial value of `WordFeatureExtractorFactory`. How about remove `new NgramExtractorTransform.NgramExtractorArguments()` in `internal INgramExtractorFactoryFactory WordFeatureExtractorFactory = new NgramExtractorTransform.NgramExtractorArguments();`?